allium/lib/diversity_index.py

```
@@ -0,0 +1,261 @@
+    """Human wording for the volume-cap cohort anchor used in tooltips.
+
+    With fewer than VOLUME_CAP_RANK operators, _rank_n_value() falls back to
+    the smallest available value, so describing it as the "5th-largest"
+    would be wrong — call it "largest available" instead.
+    """
+    if n_operators >= VOLUME_CAP_RANK:
+        return f"today's {VOLUME_CAP_RANK}th-largest"
+    return "today's largest available"
+
+
+def _mode(counter):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_mode`

(ANN202)